activation/atxdb.go

```diff
@@ -398,7 +407,7 @@ func (db *DB) SyntacticallyValidateAtx(atx *types.ActivationTx) error {
 		}
 	}
 
-	if atx.PositioningATX != *types.EmptyATXID {
+	if atx.PositioningATX != db.goldenATXID {
```

Comment:
This `if` should be updated.

An ATX with the golden ATX as the positioning ATX must have publication epoch 1 (not 0 or any other epoch - it currently uses `IsGenesis()` which is wrong).

The error message is also wrong: `no positioning atx found` should be something like `Golden ATX used for ATX in epoch X, but is only valid in epoch 0`.